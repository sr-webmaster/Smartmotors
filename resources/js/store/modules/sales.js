import makeRestStore from '@freshinup/core-ui/src/store/utils/makeRestStore'

const MODULE_URL = '/sales/v1'
export const _getItemPath = ({ id }) => `${MODULE_URL}/sales/${id}`

export default ({ items, item }, options) => {
  const store = makeRestStore('sales', { items, item }, {
    itemsPath: () => `${MODULE_URL}/sales`,
    itemPath: _getItemPath
  }, options)

  return {
    namespaced: true,
    modules: {
      export: makeRestStore(
        'export',
        { items, item },
        {
          itemsPath: () => `${MODULE_URL}/export`,
          itemPath: () => `${MODULE_URL}/export`
        }
      )
    },
    ...store
  }
}
