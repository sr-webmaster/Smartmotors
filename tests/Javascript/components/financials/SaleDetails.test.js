import { mount } from '@vue/test-utils'
import Component from '~/components/financials/SaleDetails.vue'
import { FIXTURE_ELASTICSEARCH_DEAL } from 'tests/__data__/sales'

describe('SaleDetails', () => {
  describe('Snapshot', () => {
    test('default', async () => {
      const wrapper = mount(Component)
      await wrapper.vm.$nextTick()
      expect(wrapper.element).toMatchSnapshot()
    })
  })
  test('with mocked data', async () => {
    const wrapper = mount(Component, {
      propsData: {
        pbsData: FIXTURE_ELASTICSEARCH_DEAL
      }
    })
    await wrapper.vm.$nextTick()
    expect(wrapper.find('.f-saleDetails_salePrice').text()).toEqual('$33,145.00')
    expect(wrapper.find('.f-saleDetails_listPrice').text()).toEqual('$31,490.00')
    expect(wrapper.element).toMatchSnapshot()
  })
})
