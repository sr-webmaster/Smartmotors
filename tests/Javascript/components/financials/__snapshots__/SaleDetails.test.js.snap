// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SaleDetails Snapshot default 1`] = `
<div
  class="v-card v-sheet theme--light"
>
  <div
    class="v-card__text"
  >
    <h5>
      Final Sale Price:
    </h5>
     
    <div
      class="v-alert text-xs-center headline font-weight-bold f-saleDetails_salePrice info"
    >
      <div>
        
      
    
      </div>
    </div>
     
    <h5>
      List Sale Price:
    </h5>
     
    <div
      class="v-alert text-xs-center headline font-weight-bold f-saleDetails_listPrice secondary"
    >
      <div>
        
      
    
      </div>
    </div>
  </div>
</div>
`;

exports[`SaleDetails with mocked data 1`] = `
<div
  class="v-card v-sheet theme--light"
>
  <div
    class="v-card__text"
  >
    <h5>
      Final Sale Price:
    </h5>
     
    <div
      class="v-alert text-xs-center headline font-weight-bold f-saleDetails_salePrice info"
    >
      <div>
        
      $33,145.00
    
      </div>
    </div>
     
    <h5>
      List Sale Price:
    </h5>
     
    <div
      class="v-alert text-xs-center headline font-weight-bold f-saleDetails_listPrice secondary"
    >
      <div>
        
      $31,490.00
    
      </div>
    </div>
  </div>
</div>
`;
